src/App.tsx

```diff
@@ -1,61 +1,126 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { FormEventHandler, useState } from 'react';
+import { TodoList } from './components/TodoList/TodoList';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { User } from './types/user';
+import { Todo } from './types/todo';
+import { Form } from './types/form';
+
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
+
+export const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
+
+const DEFAULT_FORM = {
+  title: '',
+  userId: 0,
+};
 
 export const App = () => {
+  const [toDoList, setToDoList] = useState<Todo[]>(todos);
+  const [data, setData] = useState<Form>(DEFAULT_FORM);
+  const [titleError, setTitleError] = useState<boolean>(false);
+  const [userSelectError, setUserSelectError] = useState<boolean>(false);
+
+  const handleInputChange:
+  React.ChangeEventHandler<HTMLInputElement> = (event) => {
+    setData({ ...data, title: event.target.value });
+    setTitleError(false);
+  };
+
+  const handleSelectChange:
+  React.ChangeEventHandler<HTMLSelectElement> = (event) => {
+    setData({ ...data, userId: +event.target.value });
```

Comment:
same as above.
Also check if `event.target.value ` is a number. use `Number` or `parseInt`